### PR TITLE
fix: buttons' border radius, width, padding and colors

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes button block support for:

 - Border radius
 - Text color
 - Background color
 - Padding
 - Width
 
| Editor | Rendered markup |
| --- | --- |
|  image |  image |

### How to test the changes in this Pull Request:

1. Draft a new newsletter and add 2 button blocks
2. Style one of the button blocks with custom values for all the style attributes listed above
3. Save and click to preview the newsletter
4. Confirm the preview markup reflects the editor styles for both the default styles and the edited one
5. Send the newsletter and confirm it also renders as expected

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/class-newspack-newsletters-renderer.php [hunk 1]
```diff
@@ -224,6 +224,30 @@ private static function get_colors( $block_attrs ) {
...
+	 */
+	private static function get_spacing_value( $value ) {
+		$presets = [
+			'50' => 'clamp( 1.25rem, 1rem + 0.8333vw, 1.5rem )',
+			'60' => 'clamp( 1.5rem, 0.75rem + 2.5vw, 2.25rem )',
+			'70' => 'clamp( 1.75rem, 0.12rem + 5.4333vw, 3.38rem )',
+			'80' => 'clamp( 2rem, -1.06rem + 10.2vw, 5.06rem )',
+		];
+		if ( 0 === strpos( $value, 'var' ) ) {
+			$preset = end( explode( '|', $value ) );
```
Comment (dkoo) [inline includes/class-newspack-newsletters-renderer.php:242]: Nit: the use of the `explode` function inside `end` is causing a PHP notice.

```suggestion
			$preset_key = explode( '|', $value );
			$preset     = end( $preset_key );
```

**Review by dkoo — CHANGES_REQUESTED** [review]
All is working as described except for the width attribute. The button doesn't appear consistent in the editor vs. in the preview and the sent email. This is true of any width setting.

In the editor, at 50%:

Screenshot 2023-09-13 at 12 11 02 PM

In the sent email:

Screenshot 2023-09-13 at 12 11 09 PM

**Comment by miguelpeixe** [comment]
> All is working as described except for the width attribute. The button doesn't appear consistent in the editor vs. in the preview and the sent email. This is true of any width setting.

That seems like a limitation with MJML when inheriting styles from the parent column layout. When adding a single button the width works well, see `Button #3` in the example below:

| Editor | Rendered |
| --- | --- |
| image | image |

Not sure how to overcome this since there are no classes to allow a custom CSS to tweak this...

**Comment by dkoo** [comment]
@miguelpeixe I think I got it working: https://github.com/Automattic/newspack-newsletters/pull/1285

The width, if defined, should be applied to the button's `mj-column` container. If not defined, then we still need to manually define a width, otherwise MJML will default to 50%. Let me know if that tweak works.

**Review by dkoo — APPROVED** [review]
Looking good to me now!

**Comment by miguelpeixe** [comment]
@dkoo I merged with the co-authoring message but it seems it didn't work for some reason 😕

**Pull request merged**